src/modules/backupRestore/backupRestoreEntities/entities/languages.ts

```diff
@@ -15,21 +15,33 @@ export const languagesEntity = {
   fetchEntities: client => client.listLanguages().toAllPromise().then(res => res.data.items.map(l => l._raw)),
   serializeEntities: languages => JSON.stringify(languages),
   deserializeEntities: serialized => JSON.parse(serialized),
-  importEntities: async (client, entities, context) => {
+  importEntities: async (client, { entities, context, options }) => {
     const importDefaultLanguage = entities.find(l =>
       l.id === defaultLanguageId
     ) as LanguageContracts.ILanguageModelContract;
     const projectDefaultLangauge = await client.viewLanguage().byLanguageId(defaultLanguageId).toPromise().then(res =>
       res.data
     );
 
+    const importLanguages = options?.excludeInactiveLanguages
+      ? entities.filter(l => l.id !== defaultLanguageId && l.is_active)
```

Comment:
It's just a detail, but you can merge the two branches into one. Sorting in the branch where there are only active languages doesn't matter and you can add into the filter a condition like this to leave the inactive languages there if `excludeInactiveLanguages` is false. `l.id !== defaultLanguageId && (!options?.excludeInactiveLanguages || l.is_active)`.

Reply:
you are right, I have force-pushed a rewritten solution :)